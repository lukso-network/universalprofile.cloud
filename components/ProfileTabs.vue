<script setup lang="ts">
type Props = {
  tabs?: ProfileViewTab[]
  activeTab?: FiltersAssetGroup
}

type Emits = (event: 'activate-tab', value: ProfileViewTab) => void

defineProps<Props>()
const emits = defineEmits<Emits>()
const { formatMessage } = useIntl()
const { isEditingGrid, isConnected } = storeToRefs(useAppStore())

const handleToggleGridEditMode = () => {
  isEditingGrid.value = !isEditingGrid.value
}
</script>

<template>
  <div
    class="mb-4 flex min-h-6 items-center justify-between border-b border-b-neutral-90 pb-4"
  >
    <ul class="flex justify-center gap-6 sm:justify-start">
      <ProfileTabsItem
        v-for="tab in tabs"
        :key="tab.id"
        :label="formatMessage(`profile_tab_${tab.id}`)"
        :count="tab.count"
        :is-active="tab.id === activeTab"
        @click="emits('activate-tab', tab)"
      />
    </ul>
    <lukso-icon
      v-if="activeTab === 'grid' && isConnected"
      name="edit"
      @click="handleToggleGridEditMode"
      class="cursor-pointer transition hover:opacity-60"
    ></lukso-icon>
  </div>
</template>
