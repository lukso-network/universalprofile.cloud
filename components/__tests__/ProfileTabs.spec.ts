import { renderSuspended } from '@nuxt/test-utils/runtime'
import { describe, it, expect, beforeEach } from 'vitest'

import ProfileTabs from '../ProfileTabs.vue'

describe('ProfileTabs', () => {
  beforeEach(() => {
    useIntl().setupIntl(defaultConfig)
  })

  it('renders default state', async () => {
    const component = await renderSuspended(ProfileTabs, {
      props: {
        tabs: [
          {
            id: 'tokens',
            count: 0,
          },
          {
            id: 'collectibles',
            count: 2,
          },
        ],
      },
    })
    expect(component.html()).toMatchSnapshot()
  })

  it('renders active state', async () => {
    const component = await renderSuspended(ProfileTabs, {
      props: {
        tabs: [
          {
            id: 'tokens',
            count: 0,
          },
          {
            id: 'collectibles',
            count: 2,
          },
        ],
        activeTab: 'collectibles',
      },
    })
    expect(component.html()).toMatchSnapshot()
  })
})
