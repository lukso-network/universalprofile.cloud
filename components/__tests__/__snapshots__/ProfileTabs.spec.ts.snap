// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileTabs > renders active state 1`] = `
"<div id="test-wrapper">
  <ul class="mb-4 flex gap-6 border-b border-b-neutral-90 pb-4">
    <li class="heading-inter-17-semi-bold flex items-center cursor-pointer opacity-50 transition hover:opacity-100">Tokens
      <!--v-if-->
    </li>
    <li class="heading-inter-17-semi-bold flex items-center cursor-default">Collectibles <span class="paragraph-inter-10-semi-bold ml-2 rounded-4 border border-neutral-20 px-1 py-[1px] bg-neutral-20 text-neutral-100">2</span></li>
  </ul>
</div>"
`;

exports[`ProfileTabs > renders default state 1`] = `
"<div id="test-wrapper">
  <ul class="mb-4 flex gap-6 border-b border-b-neutral-90 pb-4">
    <li class="heading-inter-17-semi-bold flex items-center cursor-pointer opacity-50 transition hover:opacity-100">Tokens
      <!--v-if-->
    </li>
    <li class="heading-inter-17-semi-bold flex items-center cursor-pointer opacity-50 transition hover:opacity-100">Collectibles <span class="paragraph-inter-10-semi-bold ml-2 rounded-4 border border-neutral-20 px-1 py-[1px] bg-transparent text-neutral-20">2</span></li>
  </ul>
</div>"
`;
